conduit.go

```diff
@@ -116,7 +116,8 @@ var ConnectService = &cobra.Command{
 		select {
 		case <-shutdown:
 			return nil
-		case <-finish:
+		case exitCode := <-finish:
+			os.Exit(exitCode)
```

Comment:
Is there a gotcha here from exiting early if someone comes along and wants to use the  `PostRunXXXX` family of things? ... not a big deal given the scope of this app, but if it's not _that_ important that the exact error code returned to caller, maybe wrapping/returning an `error` as "cmd xxx exited with status yyy" would give more infomation while not changing cobra's hooks stuff

Reply:
I think returning the same exit code is good user experience, but you're correct that this does interfere with the cobra pre/post

I could:

- Return an error type with an additional exit code
- Check the type in`plugin.go` around line 50
- Exit 0 or `err.ExitCode` depending

How does that suit?

Reply:
yeah pulling exit code out of an error type sounds more "correct" that exiting early :+1: